### ISSUE TITLE
Bootstrap mechanism for initial on-chain state

### ISSUE THREAD

**Issue by jynnantonix** [wormhole-foundation/wormhole#1883]
There have been a lot of transfers since the inception of the portal bridge and the initial state for accounting is going to be much larger than the maximum transaction size for cosmwasm contracts.  So we can't use the normal `instantiate` method for initializing the state.

Instead we need to add a "bootstrap" ExecuteMsg to build up the on-chain state in "max transaction size" chunks.  In the initial `instantiate` message, we will send a list of the hashes of the bootstrap messages.  Then we will send each bootstrap message in separate transactions.  While the contract is in its bootstrapping state, no other execute methods will be allowed.  Once the contract has received all the bootstrap messages from the `instantiate` message list, it will permanently switch over into the activated state.  At this point, only non-boostrap execute messages will be allowed.

We have a couple of choices for the permanent switch:

* Keep track of it as a value on-chain and check it for every execute message.
* Use a completely different contract for the initial bootstrap that uses the same on-chain data structures and then upgrade to the actual contract once all the on-chain state has been initialized.

My preference is for the latter since it more closely matches the intent and is much cleaner in the code itself.

**Comment by kcsongor** [comment]
Concur, the second option is much cleaner.

**Comment by jynnantonix** [comment]
Actually as I was implementing this I realized the "bootstrap" method works basically the same way as the "backfill" method we need for #1884 .  So rather than treating the bootstrapping phase separately, we can treat it like a special case of backfilling.

Then we don't need a separate contract and we don't need to program in a "bootstrapping" phase.  Nothing would prevent people from calling the regular methods but they might return errors until we've finished backfilling all the previous transfers.
